includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1762,6 +1762,13 @@ public function update_is_platform_checkout_enabled( $is_platform_checkout_enabl
 			$this->update_option( 'platform_checkout', $is_platform_checkout_enabled ? 'yes' : 'no' );
 			if ( ! $is_platform_checkout_enabled ) {
 				Platform_Checkout_Order_Status_Sync::remove_webhook();
+			} elseif ( WC_Payments_Features::is_upe_legacy_enabled() ) {
+				$this->update_option( WC_Payments_Features::UPE_FLAG_NAME, '0' );
+				$this->update_option( WC_Payments_Features::UPE_SPLIT_FLAG_NAME, '1' );
```

Comment:
```suggestion
				update_option( WC_Payments_Features::UPE_FLAG_NAME, '0' );
				update_option( WC_Payments_Features::UPE_SPLIT_FLAG_NAME, '1' );
```

As we were discussing in Slack, these options aren't being updated when using `$this->update_option`. Using the core `update_option` function seems to work, however. You were [doing that originally](https://github.com/Automattic/woocommerce-payments/pull/5614/commits/6a3a8ca160abaa33a4224d13997c1f0a338765d7). I'm not quite sure why using the local `update_option` function isn't working for this option but using the the core function should be fine.

Reply:
Resolved in 784702de89f91b61051b998fdbbb9694171ad1f6 and 91a32a00798edaf81df79cc973dc817e2658b28b. Cheers! 👍